api/http/api_deployments.go

```diff
@@ -1741,16 +1742,12 @@ func (d *DeploymentsApiHandlers) listDeviceDeployments(ctx context.Context,
 	var IDs []string
 	if byDeviceID {
 		did = r.PathParam("id")
-		if !govalidator.IsUUID(did) {
-			d.view.RenderError(w, r, ErrIDNotUUID, http.StatusBadRequest, l)
-			return
-		}
```

Comment:
we know that it is either UUID (in `govalidator.IsUUID` understanding) or the legacy id. we should add the validation for the latter format here rather than removing current one, and extend the tests accordingly.

Reply:
sorry for the stupid question, but what this check will actually give us?